storage/client.go

```diff
@@ -278,11 +278,23 @@ type newRangeReaderParams struct {
 
 type composeObjectRequest struct {
 	dstBucket     string
-	dstObject     string
-	srcs          []string
-	gen           int64
-	conds         *Conditions
+	dstObject     composeDstObject
+	srcs          []composeSrcObject
 	predefinedACL string
+	encryptionKey []byte
+	sendCRC32C    bool
```

Comment:
sending crc32c doesn't do anything, noted below.

Reply:
I was incorrect, looks like crc32c should be checked if passed in the request. Approving PR, thanks for your patience @tritone